crates/uv-resolver/src/resolver/mod.rs

```diff
@@ -569,38 +569,53 @@ impl<'a, Provider: ResolverProvider> Resolver<'a, Provider> {
 
                 // If the dist is an editable, return the version from the editable metadata.
                 if let Some((_local, metadata)) = self.editables.get(package_name) {
-                    let version = metadata.version.clone();
-                    return if range.contains(&version) {
-                        Ok(Some(ResolverVersion::Available(version)))
-                    } else {
-                        Ok(None)
-                    };
-                }
+                    let version = &metadata.version;
 
-                if let Ok(wheel_filename) = WheelFilename::try_from(url.raw()) {
-                    // If the URL is that of a wheel, extract the version.
-                    let version = wheel_filename.version;
-                    if range.contains(&version) {
-                        Ok(Some(ResolverVersion::Available(version)))
-                    } else {
-                        Ok(None)
```

Comment:
We have to get rid of this "fast path", because we now need the `Requires-Python` here.

Alternatively, we could move the `Requires-Python` check into `get_dependencies` for URL dependencies.